lib/sidekiq_unique_jobs/client/middleware.rb

```diff
@@ -44,14 +46,14 @@ def storage_method
       def old_unique_for?
         connection do |conn|
           conn.watch(payload_hash)
-          pid = conn.get(payload_hash).to_i
-          if pid == 1 || (pid == 2 && item['at'])
+          pid = conn.get(payload_hash)
+          if pid && (pid != 'scheduled' || item['at'])
```

Comment:
Code was a mess already. What I'd like to see is something more in the lines of:

```ruby
def old_unique_for?
  connection do |conn|
    conn.watch(payload_hash)
    pid = conn.get(payload_hash)
    return create_lock(conn) unless pid == 'scheduled' || item['at']
    conn.unwatch
  end
end

def create_lock(conn)
  conn.multi do
    return clear_exired_lock(conn) unless expires_at > 0
    conn.setex(payload_hash, expires_at, item['at'] ? 'scheduled' : item['jid'])
  end
end

def clear_expired_lock(conn)
  conn.del(payload_hash) # already expired
end
```

Thoughts on that?